CMakeLists.txt

```diff
@@ -167,20 +167,36 @@ endif()
 # Install
 #############################################################
 
-if (ENABLE_INSTALL)
-  install(TARGETS openstreetmap_cgimap
-          PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-          DESTINATION "bin")
+if(ENABLE_INSTALL)
+    include(GNUInstallDirs)
 
-  install(FILES openstreetmap-cgimap.1 DESTINATION share/man/man1)
+    set(CMAKE_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS
+        OWNER_READ OWNER_WRITE OWNER_EXECUTE
+        GROUP_READ GROUP_EXECUTE
+        WORLD_READ WORLD_EXECUTE)
 
-  install(FILES README DESTINATION share/doc/openstreetmap-cgimap)
+    set(CGIMAP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
 
-  if (BUILD_SHARED_LIBS)
-    install(TARGETS cgimap_core DESTINATION lib)
-    install(TARGETS cgimap_fcgi DESTINATION lib)
-    install(TARGETS cgimap_apidb DESTINATION lib)
-  endif()
+    if(BUILD_SHARED_LIBS)
+        set(CGIMAP_LIBS cgimap_core cgimap_fcgi cgimap_apidb libxml++)
+    endif()
+
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.23.0)
```

Comment:
I tested static and shared builds, but my local cmake version seems a bit too old to test the header files. What is the expected behavior here? Will be always include header files in the .deb package?

Reply:
The header files will only be included in the package if 
1) the library targets containing them are being installed (only in a shared build here) 
and 
2) the `Development` component is enabled (enabled by default)

There was actually a name mismatch in the install command file set (s/HEADER/HEADERS/) that I fixed now. That's probably why you didn't see the header files.

I changed it now so that headers won't be included by default in the standard all-in-one install/package, only when a component based installation is performed (`CPACK_DEB_COMPONENT_INSTALL` is set to `YES`). With per-component packages you'll a different .deb file for each component that you can install individually, so installing the `*-dev` header package is optional.

Reply:
Thank you for looking into this issue! Somehow, I still didn't make to generate multiple package

```
cmake .. -DBUILD_SHARED_LIBS=YES -DCPACK_DEB_COMPONENT_INSTALL=YES`
```

I'm pretty sure this is due to my cmake version 3.22.1. Now that the headers are not included by default, I'm good to merge this PR right now. I will add some further changes from test/381 branch to store Debian packages as build artifact to Debian 12 at least. Operations is waiting for fresh deb packages...